server/controllers/connectedAccounts.js

```diff
@@ -1,41 +1,38 @@
+const config = require('config');
+
 module.exports = (app) => {
   const errors = app.errors;
-  const models = app.set('models');
+  const models = app.get('models');
   const ConnectedAccount = models.ConnectedAccount;
-  const users = require('../controllers/users')(app);
+  const User = models.User;
 
   return {
-    post: (req, res, next) => {
-      const accessToken = req.body.accessToken;
-      const username = req.body.clientId;
-      var email = req.body.email;
-      if (!accessToken) {
-        return next(new errors.BadRequest('Access Token not provided'));
-      }
+    createOrUpdate: (req, res, next, accessToken, profile, emails) => {
+      var caId, user;
+      const attrs = { provider: req.params.service };
 
-      var caId;
-      if (!email) {
-        // TODO: temporarily putting an email address in to get past "email can't be null" requirement
-        email = `${username}.${req.params.service}@opencollective.com`;
-      }
-      users.getOrCreate(email, (err, user) => {
-        if (err) return next(err);
-        const attrs = { provider: req.params.service, UserId: user.id };
-        ConnectedAccount
-          // TODO should simplify using findOrCreate but need to upgrade Sequelize to have this fix:
-          // https://github.com/sequelize/sequelize/issues/4631
-          .findOne({ where: attrs})
-          .then(ca => ca || ConnectedAccount.create(attrs))
-          .then(ca => {
-            caId = ca.id;
-            return ca.update({username, secret: accessToken});
-          })
-          .then(() =>
-            res.send({
-              success: true,
-              token: user.generateConnectedAccountVerifiedToken(req.application, caId, username)}))
-          .catch(next);
-      });
+      // TODO should simplify using findOrCreate but need to upgrade Sequelize to have this fix:
+      // https://github.com/sequelize/sequelize/issues/4631
+      User.findOne({ where: { email: { $in: emails }}})
+        .then(u => u || User.create({
+          username: profile.username,
```

Comment:
this could be a problem. we expect the username to be unique in our database. If github username clashes with something in our db, it's going to fail. Let's not set their username to be same as on our side.